File: Formularios/Agregar/add_itemImpuesto.cs
using System;
using System.Windows.Forms;

namespace Centrex
{
    public partial class add_itemImpuesto
    {
        public add_itemImpuesto()
        {
            InitializeComponent();
        }

        private void add_modeloa_FormClosed(object sender, FormClosedEventArgs e)
        {
            Dispose();
        }

        private void add_modeloa_Load(object sender, EventArgs e)
        {
            // Cargo todos los items (EF)
            cmb_items.DataSource = combos_ef_extra.ComboItems();
            cmb_items.DisplayMember = "descript";
            cmb_items.ValueMember = "id_item";
            cmb_items.Text = "Seleccione un item";

            // Cargo todos los impuestos (EF)
            cmb_impuestos.DataSource = combos_ef.ComboImpuestosIVAActivos();
            cmb_impuestos.DisplayMember = "nombre";
            cmb_impuestos.ValueMember = "id_impuesto";
            cmb_impuestos.Text = "Seleccione un impuesto";

            chk_activo.Checked = true;
            if (edicion == true | borrado == true)
            {
                chk_secuencia.Enabled = false;

                {
                    var withBlock = edita_itemImpuesto;
                    cmb_items.SelectedValue = withBlock.IdItem;
                    cmb_impuestos.SelectedValue = withBlock.IdImpuesto;
                    chk_activo.Checked = withBlock.Activo;
                }
            }

            if (borrado == true)
            {
                cmb_items.Enabled = false;
                cmb_impuestos.Enabled = false;
                chk_activo.Enabled = false;
                cmd_ok.Visible = false;
                cmd_exit.Visible = false;
                Show();
                if (Interaction.MsgBox("¿Está seguro que desea borrar esta relación?", (MsgBoxStyle)((int)Constants.vbYesNo + (int)Constants.vbQuestion)) == MsgBoxResult.Yes)
                {
                    if (itemsImpuestos.borraritemImpuesto(edita_itemImpuesto) == false)
                    {
                        if (Interaction.MsgBox("Ocurrió un error al realizar el borrado de la relación, ¿desea intentar desactivarla para que no aparezca en la búsqueda?", (MsgBoxStyle)((int)MsgBoxStyle.Question + (int)MsgBoxStyle.YesNo)) == Constants.vbYes)
                        {
                            // Realizo un borrado lógico
                            if (itemsImpuestos.updateitemImpuesto(edita_itemImpuesto, borra: true) == true)
                            {
                                Interaction.MsgBox("Se ha podido realizar un borrado lógico, pero la relación no se borró definitivamente." + "\r" + "Esto posiblemente se deba a que esta relación tiene operaciones realizadas y/o dependencias por lo tanto no podrá borrarse", Constants.vbInformation);
                            }
                            else
                            {
                                Interaction.MsgBox("No se ha podido borrar la relación, consulte con el programador");
                            }
                        }
                    }
                }
                closeandupdate(this);
            }
        }

        private void cmd_exit_Click(object sender, EventArgs e)
        {
            Dispose();
        }

        private void cmd_ok_Click(object sender, EventArgs e)
        {
            if (cmb_items.Text == "Seleccione un item")
            {
                Interaction.MsgBox("El campo 'Item' es obligatorio y está vacio");
                return;
            }
            else if (cmb_impuestos.Text == "Seleccione un impuesto")
            {
                Interaction.MsgBox("El campo 'Impuestos' es obligatorio y está vacio");
                return;
            }

            var tmp = new ItemImpuestoEntity();
            tmp.IdItem = Conversions.ToInteger(cmb_items.SelectedValue);
            tmp.IdImpuesto = Conversions.ToInteger(cmb_impuestos.SelectedValue);
            tmp.Activo = chk_activo.Checked;

            if (edicion == true)
            {
                // tmp.IdItem = edita_itemImpuesto.IdItem
                // tmp.IdImpuesto = edita_impuesto.IdImpuesto
                if (itemsImpuestos.updateitemImpuesto(edita_itemImpuesto, tmp) == false)
                {
                    Interaction.MsgBox("Hubo un problema al actualizar la relación, consulte con su programador", Constants.vbExclamation);
                    closeandupdate(this);
                }
            }
            else
            {
                itemsImpuestos.additemImpuesto(tmp);
            }

            if (chk_secuencia.Checked == true)
            {
                cmb_items.Text = "Seleccione un item";
                cmb_impuestos.Text = "Seleccione un impuesto";
                chk_activo.Checked = true;
            }
            else
            {
                closeandupdate(this);
            }
        }

        private void pic_search_item_Click(object sender, EventArgs e)
        {
            string tmp;
            tmp = tabla;
            tabla = "itemsImpuestosItems";
            Enabled = false;
            My.MyProject.Forms.search.ShowDialog();
            tabla = tmp;

            // Establezco la opción del combo
            // cmb_items.SelectedIndex = cmb_items.FindString(info_item(id).descript)
            cmb_items.SelectedValue = id;
            id = 0;
        }

        private void pic_search_impuestos_Click(object sender, EventArgs e)
        {
            string tmp;
            tmp = tabla;
            tabla = "itemsImpuestosImpuestos";
            Enabled = false;
            My.MyProject.Forms.search.ShowDialog();
            tabla = tmp;

            // Establezco la opción del combo
            // cmb_impuestos.SelectedIndex = cmb_impuestos.FindString(info_impuesto(id).nombre)
            cmb_impuestos.SelectedValue = id;
            id = 0;
        }

        private void cmb_items_KeyPress(object sender, KeyPressEventArgs e)
        {
            // e.KeyChar = ""
        }

        private void cmb_impuestos_KeyPress(object sender, KeyPressEventArgs e)
        {
            // e.KeyChar = ""
        }
    }
}

File: __Viejo/Formularios/Agregar/CS/add_itemImpuesto.vb
Public Class add_itemImpuesto

    Private Sub add_modeloa_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        Me.Dispose()
    End Sub

    Private Sub add_modeloa_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        form = Me

        'Cargo todas los items
        cargar_combo(cmb_items, "SELECT id_item, descript FROM items ORDER BY item ASC", basedb, "descript", "id_item")
        cmb_items.Text = "Seleccione un item"

        'Cargo todos los impuestos
        cargar_combo(cmb_impuestos, "SELECT id_impuesto, nombre FROM impuestos ORDER BY nombre ASC", basedb, "nombre", "id_impuesto")
        cmb_impuestos.Text = "Seleccione un impuesto"
        
        chk_activo.Checked = True
        If edicion = True Or borrado = True Then
            chk_secuencia.Enabled = False

            With edita_itemImpuesto
                cmb_items.SelectedValue = .id_item
                cmb_impuestos.SelectedValue = .id_impuesto
                chk_activo.Checked = .activo
            End With
        End If

        If borrado = True Then
            cmb_items.Enabled = False
            cmb_impuestos.Enabled = False
            chk_activo.Enabled = False
            cmd_ok.Visible = False
            cmd_exit.Visible = False
            Me.Show()
            If MsgBox("¿Está seguro que desea borrar esta relación?", vbYesNo + vbQuestion) = MsgBoxResult.Yes Then
                If (borraritemImpuesto(edita_itemImpuesto)) = False Then
                    If (MsgBox("Ocurrió un error al realizar el borrado de la relación, ¿desea intectar desactivarla para que no aparezca en la búsqueda?" _
                     , MsgBoxStyle.Question + MsgBoxStyle.YesNo)) = vbYes Then
                        'Realizo un borrado lógico
                        If updateitemImpuesto(edita_itemImpuesto, , True) = True Then
                            MsgBox("Se ha podido realizar un borrado lógico, pero la realación no se borró definitivamente." + Chr(13) + _
                                "Esto posiblemente se deba a que el está relación, tiene operaciones realizadas y/o dependencias por lo tanto no podrá borrarse", vbInformation)
                        Else
                            MsgBox("No se ha podido borrar la relación, consulte con el programador")
                        End If
                    End If
                End If
            End If
            closeandupdate(Me)
        End If
    End Sub

    Private Sub cmd_exit_Click(sender As Object, e As EventArgs) Handles cmd_exit.Click
        Me.Dispose()
    End Sub

    Private Sub cmd_ok_Click(sender As Object, e As EventArgs) Handles cmd_ok.Click
        If cmb_items.Text = "Seleccione un item" Then
            MsgBox("El campo 'Item' es obligatorio y está vacio")
            Exit Sub
        ElseIf cmb_impuestos.Text = "Seleccione un impuesto" Then
            MsgBox("El campo 'Impuestos' es obligatorio y está vacio")
            Exit Sub
        End If

        Dim tmp As New itemImpuesto
        tmp.id_item = cmb_items.SelectedValue
        tmp.id_impuesto = cmb_impuestos.SelectedValue
        tmp.activo = chk_activo.Checked

        If edicion = True Then
            'tmp.id_item = edita_itemImpuesto.id_item
            'tmp.id_impuesto = edita_impuesto.id_impuesto
            If updateitemImpuesto(edita_itemImpuesto, tmp) = False Then
                MsgBox("Hubo un problema al actualizar la relación, consulte con su programdor", vbExclamation)
                closeandupdate(Me)
            End If
        Else
            additemImpuesto(tmp)
        End If

        If chk_secuencia.Checked = True Then
            cmb_items.Text = "Seleccione un item"
            cmb_items.Text = "Seleccione un impuesto"
            chk_activo.Checked = True
        Else
            closeandupdate(Me)
        End If
    End Sub

    Private Sub pic_search_item_Click(sender As Object, e As EventArgs) Handles pic_search_item.DoubleClick
        Dim tmp As String
        tmp = tabla
        tabla = "itemsImpuestosItems"
        Me.Enabled = False
        search.ShowDialog()
        tabla = tmp

        'Establezco la opción del combo
        'cmb_items.SelectedIndex = cmb_items.FindString(info_item(id).descript)
        cmb_items.SelectedValue = id
        id = 0
    End Sub

    Private Sub pic_search_impuestos_Click(sender As Object, e As EventArgs) Handles pic_search_impuestos.DoubleClick
        Dim tmp As String
        tmp = tabla
        tabla = "itemsImpuestosImpuestos"
        Me.Enabled = False
        search.ShowDialog()
        tabla = tmp

        'Establezco la opción del combo
        'cmb_impuestos.SelectedIndex = cmb_impuestos.FindString(info_impuesto(id).nombre)
        cmb_impuestos.SelectedValue = id
        id = 0
    End Sub

    Private Sub cmb_items_KeyPress(sender As Object, e As KeyPressEventArgs) Handles cmb_items.KeyPress
        'e.KeyChar = ""
    End Sub

    Private Sub cmb_impuestos_KeyPress(sender As Object, e As KeyPressEventArgs) Handles cmb_impuestos.KeyPress
        'e.KeyChar = ""
    End Sub
End Class